lms/djangoapps/email_marketing/signals.py

```diff
@@ -11,17 +11,21 @@
 from sailthru.sailthru_error import SailthruClientError
 
 from email_marketing.models import EmailMarketingConfiguration
-from lms.djangoapps.email_marketing.tasks import update_user, update_user_email
+from lms.djangoapps.email_marketing.tasks import update_user, update_user_email, get_email_cookies_via_sailthru
 from student.cookies import CREATE_LOGON_COOKIE
 from student.views import REGISTER_USER
 from util.model_utils import USER_FIELD_CHANGED
 
+from celery.exceptions import TimeoutError
+
 log = logging.getLogger(__name__)
 
 # list of changed fields to pass to Sailthru
 CHANGED_FIELDNAMES = ['username', 'is_active', 'name', 'gender', 'education',
                       'age', 'level_of_education', 'year_of_birth',
                       'country']
+# timeout value in seconds for sailthru api calls
+SAILTHRU_TIMEOUT = 3
```

Comment:
I recommend we have a better name for this:
`COOKIE_CREATION_REQUEST_TIME_LIMIT_IN_SECONDS`